kernel/driver.cc

```diff
@@ -636,7 +651,30 @@ int main(int argc, char **argv)
 			if (Tcl_EvalFile(interp, scriptfile.c_str()) != TCL_OK)
 				log_error("TCL interpreter returned an error: %s\n", Tcl_GetStringResult(yosys_get_tcl_interp()));
 #else
-			log_error("Can't exectue TCL script: this version of yosys is not built with TCL support enabled.\n");
+			log_error("Can't execute TCL script: this version of yosys is not built with TCL support enabled.\n");
+#endif
+		} else if (scriptfile_python) {
+#ifdef WITH_PYTHON
+			PyObject *sys = PyImport_ImportModule("sys");
+			PyObject *new_argv = PyList_New(argc - optind + 1);
+			PyList_SetItem(new_argv, 0, PyUnicode_FromString(scriptfile.c_str()));
+			for (int i = optind; i < argc; ++i)
+				PyList_SetItem(new_argv, i - optind + 1, PyUnicode_FromString(argv[i]));
+
+			PyObject *old_argv = PyObject_GetAttrString(sys, "argv");
+			PyObject_SetAttrString(sys, "argv", new_argv);
+			Py_DECREF(old_argv);
+
+			PyRun_SimpleString(("import os;sys.path.insert(0, os.path.dirname(os.path.abspath(\""+scriptfile+"\")))").c_str());
```

Comment:
I think the escaping issues I mentioned on the call could be avoided using something like this instead, similar how argv is already handled:

```
  PyObject *py_path = PyUnicode_FromString(scriptfile.c_str());
  PyObject_SetAttrString(sys, "_yosys_script_path", py_path);
  Py_DECREF(py_path);
  PyRun_SimpleString("import os, sys; sys.path.insert(0, path.dirname(os.path.abspath(sys._yosys_script_path)))");
```

I tested this with a small stand-alone program and had to import both `os` and `sys` in the python snippet.